app/v1/controllers/ListBsasController.scala

```diff
@@ -90,8 +96,11 @@ class ListBsasController @Inject()(val authService: EnrolmentsAuthService,
             .as(MimeTypes.JSON)
         }
       result.leftMap { errorWrapper =>
-        val correlationId = getCorrelationId(errorWrapper)
-        val result = errorResult(errorWrapper).withApiHeaders(correlationId)
+        val resCorrelationId = errorWrapper.correlationId
+        val result = errorResult(errorWrapper).withApiHeaders(resCorrelationId)
+        logger.info(
+          s"[${endpointLogContext.controllerName}][${endpointLogContext.endpointName}] - " +
+            s"Error response received with CorrelationId: $resCorrelationId")
 
         auditSubmission(
           GenericAuditDetail(
```

Comment:
Use **resCorrelationId** instead of correlationId in GenericAuditDetail. i.e. `X-CorrelationId` = resCorrelationId